### PR TITLE
improve test coverage in service pkg

### PR DESCRIPTION
**Pull request by n0npax**
fix some potential panics

### CODE REVIEW

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by josedonizetti** [comment]
@minikube-bot Ok to test

File: pkg/minikube/service/service_test.go [hunk 1]
```diff
@@ -493,4 +538,341 @@ func TestGetServiceURLsForService(t *testing.T) {
...
+			description: "correctly return empty serviceURLs",
+			namespace:   "default",
+			service:     "mock-dashboard-no-ports",
+			api:         defaultAPI,
+			expected:    []string{},
+			err:         true,
+		},
+	}
+	defer revertK8sClient(K8s)
+	for _, mode := range []bool{true, false} {
```
Comment (josedonizetti) [inline pkg/minikube/service/service_test.go]: This nested `for` test is really hard to read. Can you break it into an array of structures, and then do only one `for` to test it?

Comment (n0npax) [reply pkg/minikube/service/service_test.go]: fixed. Should be better now.

File: pkg/minikube/service/service_test.go [hunk 2]
```diff
@@ -493,4 +538,358 @@ func TestGetServiceURLsForService(t *testing.T) {
...
+			err:            true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			mockK8sConfigByte := []byte(test.config)
+			mockK8sConfigPath := test.kubeconfigPath
+			err := ioutil.WriteFile(mockK8sConfigPath, mockK8sConfigByte, 0644)
+			defer func() { os.Remove(mockK8sConfigPath) }()
```
Comment (josedonizetti) [inline pkg/minikube/service/service_test.go]: `defer os.Remove(mockK8sConfigPath)`

File: pkg/minikube/service/service_test.go [hunk 3]
```diff
@@ -33,30 +37,68 @@ import (
...
+	secretsMap   map[string]typed_core.SecretInterface
+	Fake         fake.FakeCoreV1
 }
 
+func init() {
+	getCoreClientFail = false
+}
+
+// Force GetCoreClient to fail
+var getCoreClientFail bool
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/service/service_test.go]: fixed. Thanks!

File: pkg/minikube/service/service.go [hunk 4]
```diff
@@ -310,7 +314,11 @@ func GetServiceListByLabel(namespace string, key string, value string) (*core.Se
 	if err != nil {
 		return &core.ServiceList{}, &retry.RetriableError{Err: err}
 	}
-	return getServiceListFromServicesByLabel(client.Services(namespace), key, value)
+	svc := client.Services(namespace)
+	if svc == nil {
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/service/service.go]: fixed for secrets & services. Thanks!

File: pkg/minikube/service/service_test.go (hunk 2 above)
Comment (n0npax) [reply pkg/minikube/service/service_test.go]: fixed

File: pkg/minikube/service/service_test.go [hunk 5]
```diff
@@ -493,4 +530,316 @@ func TestGetServiceURLsForService(t *testing.T) {
...
+		}
+		defer revertK8sClient(K8s)
+		if err := DeleteSecret(ns, secret); err != nil {
+			t.Fatalf("DeleteSecret not expected to fail for namespace: %v and secret %v but got err: %v", ns, secret, err)
+		}
+	})
+}
+
+func TestCreateSecret(t *testing.T) {
+	t.Run("failed Get Client", func(t *testing.T) {
```
Comment (tstromberg) [inline pkg/minikube/service/service_test.go]: Can this be refactored to use a table driven test?

https://github.com/golang/go/wiki/TableDrivenTests

Perhaps by adding a bool field for whether an error is expected?

Comment (n0npax) [reply pkg/minikube/service/service_test.go]: ok. Is table-driven approach preferred even if there are just 2 tests?

**Comment by n0npax** [comment]
Thanks for review and approve.

**Pull request merged**